puppet/modules/site_config/manifests/remove/files.pp

```diff
@@ -31,10 +32,6 @@
       path => '/var/log/',
       recurse => true,
       matches => 'leap_mx*';
-    'mx':
```

Comment:
The problem with deleting this block is that we have systems that need to be cleaned up, for example this machine (which doesn't exist now, but this happened there):

```
root@chipmonk:/var/log/leap# ls -altr
total 98524
-rw-r-----  1 root adm    30318 Apr 10 06:40 mx.log.7.gz
-rw-r-----  1 root adm     7607 Apr 28 06:34 mx.log.6.gz
-rw-r-----  1 root adm  1000035 Apr 28 23:37 mx.log.98
-rw-r-----  1 root root 2960713 May 14 12:42 mx.log.97
-rw-r-----  1 root root 1000097 May 21 15:01 mx.log.96
-rw-r-----  1 root root 1000025 May 25 14:52 mx.log.95
-rw-r-----  1 root root 1000149 Jun  4 07:47 mx.log.94
-rw-r-----  1 root root 1000026 Jun 11 20:38 mx.log.93
-rw-r-----  1 root root 1000023 Jun 17 15:27 mx.log.92
-rw-r-----  1 root root 1001068 Jun 20 07:28 mx.log.91
-rw-r-----  1 root root 1000934 Jun 21 05:58 mx.log.90
-rw-r-----  1 root root 1001807 Jun 22 05:28 mx.log.89
-rw-r-----  1 root root 1000028 Jun 22 19:47 mx.log.88
-rw-r-----  1 root root 1000994 Jun 23 08:58 mx.log.87
-rw-r-----  1 root root 1000020 Jun 23 20:28 mx.log.86
-rw-r-----  1 root root 1000543 Jun 24 06:58 mx.log.85
-rw-r-----  1 root root 1001271 Jun 24 17:58 mx.log.84
-rw-r-----  1 root root 1001720 Jun 25 04:58 mx.log.83
-rw-r-----  1 root root 1001782 Jun 25 15:58 mx.log.82
-rw-r-----  1 root root 1001262 Jun 26 02:28 mx.log.81
-rw-r-----  1 root root 1000017 Jun 26 13:58 mx.log.80
-rw-r-----  1 root root 1000118 Jun 27 00:28 mx.log.79
-rw-r-----  1 root root 1000091 Jun 27 11:28 mx.log.78
-rw-r-----  1 root root 1001261 Jun 27 22:28 mx.log.77
-rw-r-----  1 root root 1001129 Jun 28 08:58 mx.log.76
-rw-r-----  1 root root 1000014 Jun 28 19:28 mx.log.75
-rw-r-----  1 root root 1001804 Jun 29 06:28 mx.log.74
-rw-r-----  1 root root 1000083 Jun 29 16:28 mx.log.73
-rw-r-----  1 root root 1000022 Jun 30 02:58 mx.log.72
-rw-r-----  1 root root 1000956 Jun 30 12:58 mx.log.71
-rw-r-----  1 root root 1001579 Jun 30 20:28 mx.log.70
-rw-r-----  1 root root 1000925 Jul  1 03:28 mx.log.69
-rw-r-----  1 root root 1000542 Jul  1 10:28 mx.log.68
-rw-r-----  1 root root 1001619 Jul  1 17:28 mx.log.67
-rw-r-----  1 root root 1000058 Jul  2 00:28 mx.log.66
-rw-r-----  1 root root 1001211 Jul  2 07:28 mx.log.65
-rw-r-----  1 root root 1001901 Jul  2 14:28 mx.log.64
-rw-r-----  1 root root 1001299 Jul  2 21:28 mx.log.63
-rw-r-----  1 root root 1001535 Jul  3 04:28 mx.log.62
-rw-r-----  1 root root 1001657 Jul  3 11:28 mx.log.61
-rw-r-----  1 root root 1000338 Jul  3 18:28 mx.log.60
-rw-r-----  1 root root 1000089 Jul  4 01:28 mx.log.59
-rw-r-----  1 root root 1000153 Jul  4 07:58 mx.log.58
-rw-r-----  1 root root 1000834 Jul  4 14:28 mx.log.57
-rw-r-----  1 root root 1000042 Jul  4 20:28 mx.log.56
-rw-r-----  1 root root 1000375 Jul  5 01:58 mx.log.55
-rw-r-----  1 root root 1000072 Jul  5 06:58 mx.log.54
-rw-r-----  1 root root 1000133 Jul  5 12:28 mx.log.53
-rw-r-----  1 root root 1001202 Jul  5 17:58 mx.log.52
-rw-r-----  1 root root 1000453 Jul  5 23:28 mx.log.51
-rw-r-----  1 root root 1001891 Jul  6 04:28 mx.log.50
-rw-r-----  1 root root 1000815 Jul  6 09:58 mx.log.49
-rw-r-----  1 root root 1001172 Jul  6 15:28 mx.log.48
-rw-r--r--  1 root root    2261 Jul  6 18:15 deploy-summary.log
-rw-r--r--  1 root root  356803 Jul  6 18:15 deploy.log
-rw-r-----  1 root root 1000479 Jul  6 18:43 mx.log.47
-rw-r-----  1 root root 1001000 Jul  6 19:19 mx.log.46
-rw-r-----  1 root root 1001050 Jul  6 19:55 mx.log.45
-rw-r-----  1 root root 1000547 Jul  6 20:28 mx.log.44
-rw-r-----  1 root root 1000554 Jul  6 21:04 mx.log.43
-rw-r-----  1 root root 1000920 Jul  6 21:39 mx.log.42
-rw-r-----  1 root root 1000163 Jul  6 22:14 mx.log.41
-rw-r-----  1 root root 1001052 Jul  6 22:49 mx.log.40
-rw-r-----  1 root root 1000978 Jul  6 23:26 mx.log.39
-rw-r-----  1 root root 1000862 Jul  6 23:58 mx.log.38
-rw-r-----  1 root root 1000589 Jul  7 00:32 mx.log.37
-rw-r-----  1 root root 1000173 Jul  7 01:07 mx.log.36
-rw-r-----  1 root root 1000095 Jul  7 01:33 mx.log.35
-rw-r-----  1 root root 1000508 Jul  7 02:08 mx.log.34
-rw-r-----  1 root root 1000083 Jul  7 02:40 mx.log.33
-rw-r-----  1 root root 1000986 Jul  7 03:15 mx.log.32
-rw-r-----  1 root root 1000946 Jul  7 03:48 mx.log.31
-rw-r-----  1 root root 1000900 Jul  7 04:21 mx.log.30
-rw-r-----  1 root root 1000990 Jul  7 04:57 mx.log.29
-rw-r-----  1 root root 1001195 Jul  7 05:28 mx.log.28
-rw-r-----  1 root root 1002261 Jul  7 05:58 mx.log.27
-rw-r-----  1 root root 1001019 Jul  7 06:28 mx.log.26
drwxr-xr-x 14 root root   32768 Jul  7 06:54 ..
-rw-r-----  1 root root 1000011 Jul  7 06:58 mx.log.25
-rw-r-----  1 root root 1000446 Jul  7 07:28 mx.log.24
-rw-r-----  1 root root 1000240 Jul  7 07:58 mx.log.23
-rw-r-----  1 root root 1000097 Jul  7 08:28 mx.log.22
-rw-r-----  1 root root 1001366 Jul  7 08:58 mx.log.21
-rw-r-----  1 root root 1000680 Jul  7 09:28 mx.log.20
-rw-r-----  1 root root 1000825 Jul  7 09:58 mx.log.19
-rw-r-----  1 root root 1000516 Jul  7 10:28 mx.log.18
-rw-r-----  1 root root 1000928 Jul  7 10:58 mx.log.17
-rw-r-----  1 root root 1000379 Jul  7 11:28 mx.log.16
-rw-r-----  1 root root 1000091 Jul  7 11:58 mx.log.15
-rw-r-----  1 root root 1001663 Jul  7 12:28 mx.log.14
-rw-r-----  1 root root 1000185 Jul  7 12:58 mx.log.13
-rw-r-----  1 root root 1001815 Jul  7 13:28 mx.log.12
-rw-r-----  1 root root 1000113 Jul  7 13:58 mx.log.11
-rw-r-----  1 root root 1000039 Jul  7 14:28 mx.log.10
-rw-r-----  1 root root 1000081 Jul  7 14:58 mx.log.9
-rw-r-----  1 root root 1000306 Jul  7 15:28 mx.log.8
-rw-r-----  1 root root 1001058 Jul  7 15:57 mx.log.7
-rw-r-----  1 root root 1000454 Jul  7 16:27 mx.log.6
-rw-r-----  1 root root 1000935 Jul  7 16:57 mx.log.5
-rw-r-----  1 root root 1000841 Jul  7 17:27 mx.log.4
-rw-r-----  1 root root 1001890 Jul  7 17:58 mx.log.3
-rw-r-----  1 root root 1002052 Jul  7 18:28 mx.log.2
-rw-r-----  1 root root 1000710 Jul  7 18:57 mx.log.1
drwxr-x---  2 root adm     4096 Jul  7 18:57 .
-rw-r-----  1 root root  526735 Jul  7 19:11 mx.log
```

Since we are rotating 5 entries (in the augeas resource in leap::logfile, line 17), so we should only have the following logfiles in that directory ever: mx.log, mx.log.[1-5], with an optional .gz suffix. I think the following would take care of cleaning up the junk from before:

```
-matches => 'mx.log*';
+matches => [ 'mx.log.[6-9](.gz)?', 'mx.log.[0-9][0-9](.gz)?'];
```

here is my test demonstrating this:

```
micah@muck:mx_log_test$ for file in `seq 1 98`; do touch mx.log.$file; done
micah@muck:mx_log_test$ touch deploy-summary.log deploy.log mx.log mx.log.6.gz mx.log.7.gz
micah@muck:mx_log_test$ ls
deploy.log          mx.log.12  mx.log.18  mx.log.23  mx.log.29  mx.log.34  mx.log.4   mx.log.45  mx.log.50  mx.log.56  mx.log.61  mx.log.67    mx.log.71  mx.log.77    mx.log.81  mx.log.87  mx.log.92  mx.log.98
deploy-summary.log  mx.log.13  mx.log.19  mx.log.24  mx.log.3   mx.log.35  mx.log.40  mx.log.46  mx.log.51  mx.log.57  mx.log.62  mx.log.68    mx.log.72  mx.log.78    mx.log.82  mx.log.88  mx.log.93
mx.log              mx.log.14  mx.log.2   mx.log.25  mx.log.30  mx.log.36  mx.log.41  mx.log.47  mx.log.52  mx.log.58  mx.log.63  mx.log.69    mx.log.73  mx.log.79    mx.log.83  mx.log.89  mx.log.94
mx.log.1            mx.log.15  mx.log.20  mx.log.26  mx.log.31  mx.log.37  mx.log.42  mx.log.48  mx.log.53  mx.log.59  mx.log.64  mx.log.6.gz  mx.log.74  mx.log.7.gz  mx.log.84  mx.log.9   mx.log.95
mx.log.10           mx.log.16  mx.log.21  mx.log.27  mx.log.32  mx.log.38  mx.log.43  mx.log.49  mx.log.54  mx.log.6   mx.log.65  mx.log.7     mx.log.75  mx.log.8     mx.log.85  mx.log.90  mx.log.96
mx.log.11           mx.log.17  mx.log.22  mx.log.28  mx.log.33  mx.log.39  mx.log.44  mx.log.5   mx.log.55  mx.log.60  mx.log.66  mx.log.70    mx.log.76  mx.log.80    mx.log.86  mx.log.91  mx.log.97
micah@muck:mx_log_test$ ls | egrep -v '(mx.log.[6-9](.gz)?|mx.log.[0-9][0-9](.gz)?)'
deploy.log
deploy-summary.log
mx.log
mx.log.1
mx.log.2
mx.log.3
mx.log.4
mx.log.5
micah@muck:mx_log_test$ 
```